blockly-core/core/block_space.js

```diff
@@ -486,3 +492,58 @@ Blockly.BlockSpace.prototype.remainingCapacity = function() {
   }
   return this.maxBlocks - this.getAllBlocks().length;
 };
+
+/**
+* Make a list of all the delete areas for this blockSpace.
+*/
+Blockly.BlockSpace.prototype.recordDeleteAreas = function() {
+  this.deleteAreas_ = [];
+
+  if (this.trashcan) {
+    goog.array.extend(this.deleteAreas_, this.trashcan.getRect());
+    this.deleteAreaTrash_ = this.trashcan.getRect();
+  } else {
+    this.deleteAreaTrash_ = null;
+  }
+
+  if (this.flyout_) {
+    goog.array.extend(this.deleteAreas_, this.flyout_.getRect());
+  }
+
+  if (this.blockSpaceEditor) {
+    goog.array.extend(this.deleteAreas_,
+        this.blockSpaceEditor.getDeleteAreas());
+  }
+};
+
+/**
+* Is the mouse event over a delete area (toolbar or non-closing flyout)?
+* Opens or closes the trashcan and sets the cursor as a side effect.
+* @param {!Event} e Mouse move event.
+* @return {boolean} True if event is in a delete area.
+*/
+Blockly.BlockSpace.prototype.isDeleteArea = function(e) {
```

Comment:
Do we need to worry about isDeleteArea being called before deleteAreas were recorded?